airflow/providers/ftp/example_dags/example_ftp.py

```diff
@@ -0,0 +1,51 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from datetime import datetime
+
+from airflow import models
+from airflow.providers.ftp.operators.ftp import FTPOperation, FTPOperator
+
+with models.DAG(
+    "example_ftp_put_get",
+    schedule_interval=None,
+    start_date=datetime(2021, 1, 1),
+    catchup=False,
+) as dag:
+    # [START howto_ftp_put]
+    ftp_put = FTPOperator(
+        task_id="test_ftp_put",
+        ftp_conn_id="ftp_default",
+        local_filepath="/tmp/filepath",

```

Comment:
This should be a transfer operator.
See examples for LocalTo* operators.
Since operators dont share disk the source of temp folder doesnt seem right.

Reply:
Do you mean that the `local_filepath` should not be set to `/tmp/filepath`? I was looking at the example given for SFTP put [here](https://airflow.apache.org/docs/apache-airflow-providers-sftp/stable/_api/airflow/providers/sftp/operators/sftp/index.html), and it seems that they put a similar path. Do you mean that the `local_filepath` should instead be something like "relative/path/to/file" like from [here](https://airflow.apache.org/docs/apache-airflow-providers-amazon/3.3.0/operators/transfer/local_to_s3.html)?

Reply:
my comment is on the whole operator
check for example https://github.com/apache/airflow/blob/219df18414185e426aab9bb8f297785d6f54f6a5/airflow/providers/microsoft/azure/transfers/local_to_wasb.py#L29

Reply:
I see, so do you mean that rather than defining the FTP Operator within `airflow.providers.ftp.operators`, instead define them in a class within the `airflow.providers.ftp.transfers` folder? Looking through examples of operators defined in transfers folder, like for the microsoft azure transfer operators, it seems like many of them are unidirectional in the sense that they only upload files from local to a remote server, or only download files from a remote to a local. I was thinking that maybe it would make more sense to make FTPOperator a pure operator rather than a transfer operator since it can do bidirectional file movements and the direction is determined by passed-in parameters from the user, similar to how the SFTPOperator is defined  [here](https://github.com/apache/airflow/tree/219df18414185e426aab9bb8f297785d6f54f6a5/airflow/providers/sftp/operators).

Reply:
I was wondering if it would be more appropriate to define the FTPOperator similarly to SFTPOperator, rather than having it as a specific transfer operator?

Reply:
I think we have two different approaches for that. And we can decide either way. I personally think the "Local" is a bit artifficial for transfer. Having to define connection to download/upload files is not necessarily what you woul dlike ot do. 

I think we are not currently in the spree of cleaning and unifying this, so I would be ok with merging it as is.

WDYT @eladkal

Reply:
I think consistancy is important. Its very hard to work with Airflow when every provider sets its own unique behavior for the same action.

That said, If we accept it as is, at lease the class name should be more explanatory. It should contain some verb/action to better describe what it does.

Reply:
That is a good point. Would a name like `FTPFileTransfer` make more sense?

Reply:
Sounds ok

Reply:
I have updated the operator to reflect this name change to `FTPFileTransfer`. Let me know what you think.

Reply:
@RachitSharma2001 can you modify the example dag to AIP-47 format?
see many PRs for reference https://github.com/apache/airflow/pulls?q=is%3Aopen+is%3Apr+label%3Aarea%3Asystem-tests

Reply:
Yeah. Would be great to get it to AIP-47 format already.

Reply:
I added the example dag mainly as reference for documentation. However, I have not set up any system tests, as that would require setting up a local FTP server, which currently has not been added to the ci. Would it still be required to move the example dags to `tests/system/providers/ftp` and change it to AIP-47 even if it isn't meant to be a system test?

Reply:
It should be, I think. All our examples should bin AIP-47 compatible form so one day we can automate running them